src/utils/tokens.ts

```diff
@@ -7,3 +11,59 @@ export const getColonyTokenAddress = async (
   const tokenAddress = await colonyClient.getToken();
   return tokenAddress;
 };
+
+export const getExistingTokenAddresses = (colony: ColonyQuery): string[] =>
+  colony.tokens?.items
+    .map((tokenItem) => tokenItem?.tokenAddress)
+    .filter((item): item is string => !!item) ?? [];
+
+export const updateColonyTokens = async (
```

Comment:
This is similar in spirit to its CDapp counterpart (can be found in editColony saga), however, more granular, so that we don't overwrite state changes that happened after the current motion was created.

Reply:
nice one taking this into consideration 👍